src/protocols/secure_channel/CASESession.cpp

```diff
@@ -404,187 +445,283 @@ CHIP_ERROR CASESession::HandleSigmaR1(System::PacketBufferHandle && msg)
     ChipLogDetail(SecureChannel, "Peer assigned session key ID %d", initiatorSessionId);
     SetPeerSessionId(initiatorSessionId);
 
-    SuccessOrExit(err = tlvReader.Next());
-    VerifyOrExit(TLV::TagNumFromTag(tlvReader.GetTag()) == ++decodeTagIdSeq, err = CHIP_ERROR_INVALID_TLV_TAG);
-    SuccessOrExit(err = tlvReader.GetBytes(destinationIdentifier, sizeof(destinationIdentifier)));
+    if (sessionResumptionRequested && memcmp(resumptionID, mResumptionId, kCASEResumptionIDSize) == 0)
```

Comment:
It should be impossible to have `sessionResumptionRequested == true` and `memcmp(resumptionID, mResumptionId, kCASEResumptionIDSize) != 0`, since resumption ID is part of the MIC. I think the resumption ID should be checked for `memcmp == 0` in `ParseSessionResumptionRequest()` instead.